python/src/aiconfig/editor/client/src/reducers/aiconfigReducer.ts

```diff
@@ -182,7 +183,11 @@ export default function aiconfigReducer(
             model: action.modelName
               ? {
                   name: action.modelName,
-                  settings: modelSettings,
+                  // TODO (rossdanlm): For now just clearing the model field whenever new model name is selected
+                  settings: {
+                    ...modelSettings,
+                    model: undefined,
```

Comment:
If we explicitly set to undefined here this is actually going to have an explicit value in the settings in the config, I think, even for settings that don't have model. Instead, can we just do this at the top when defining `modelSettings`:

```
... modelSettings = promptModel.settings;
// TODO (rossdanlm): For now just clearing the model field whenever new model name is selected
  delete modelSettings.model;
}
```